src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -2049,10 +2057,82 @@ WaitForMiliseconds(long timeout)
  * replication.
  */
 static XLogRecPtr
-GetSubscriptionPosition(MultiConnection *connection, Bitmapset *tableOwnerIds)
+GetSubscriptionPosition(MultiConnection *connection, Bitmapset *tableOwnerIds,
+						char *operationPrefix)
 {
-	char *subscriptionValueList = ShardMoveSubscriptionNamesValueList(tableOwnerIds);
+	char *subscriptionValueList = ShardSubscriptionNamesValueList(tableOwnerIds,
+																  operationPrefix);
 	return GetRemoteLSN(connection, psprintf(
 							"SELECT min(latest_end_lsn) FROM pg_stat_subscription "
 							"WHERE subname IN %s", subscriptionValueList));
 }
+
+
+/*TODO(saawasek):Refactor this for ShardMove too.*/
+void
+CreateShardSubscription(MultiConnection *connection, char *sourceNodeName,
+						int sourceNodePort, char *userName, char *databaseName,
+						char *publicationName, char *slotName,
+						Oid ownerId)
+{
+	StringInfo createSubscriptionCommand = makeStringInfo();
+	StringInfo conninfo = makeStringInfo();
+
+	/*
+	 * The CREATE USER command should not propagate, so we temporarily
+	 * disable DDL propagation.
+	 */
+	SendCommandListToWorkerOutsideTransaction(
+		connection->hostname, connection->port, connection->user,
+		list_make2(
+			"SET LOCAL citus.enable_ddl_propagation TO OFF;",
+			psprintf(
+				"CREATE USER %s SUPERUSER IN ROLE %s",
+				ShardSubscriptionRole(ownerId, SHARD_SPLIT_SUBSCRIPTION_ROLE_PREFIX),
+				GetUserNameFromId(ownerId, false)
+				)));
+
+	appendStringInfo(conninfo, "host='%s' port=%d user='%s' dbname='%s' "
+							   "connect_timeout=20",
+					 escape_param_str(sourceNodeName), sourceNodePort,
+					 escape_param_str(userName), escape_param_str(databaseName));
+
+	appendStringInfo(createSubscriptionCommand,
+					 "CREATE SUBSCRIPTION %s CONNECTION %s PUBLICATION %s "
+					 "WITH (citus_use_authinfo=true, enabled=false, create_slot=false, copy_data=false, slot_name='%s')",
+					 quote_identifier(ShardSubscriptionName(ownerId,
+															SHARD_SPLIT_SUBSCRIPTION_PREFIX)),
+					 quote_literal_cstr(conninfo->data),
+					 quote_identifier(publicationName),
+					 slotName);
+
+	ExecuteCriticalRemoteCommand(connection, createSubscriptionCommand->data);
+	pfree(createSubscriptionCommand->data);
+	pfree(createSubscriptionCommand);
+	ExecuteCriticalRemoteCommand(connection, psprintf(
```

Comment:
We should add a small comment on why we do a Create follow by Alter (explaining the permission problem)

Reply:
I think its clear that we are creating a subscription for user from the previous steps. 
Earlier, the subscriptions were directly created for super user role. But now we clearly create a new role for the corresponding table owner signifying 1-1.